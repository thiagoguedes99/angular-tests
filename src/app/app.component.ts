import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterLink, RouterLinkActive, RouterOutlet } from '@angular/router';

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [CommonModule, RouterOutlet, RouterLink, RouterLinkActive],
  // templateUrl: './app.component.html',
  template: `
    <h1>Hello, {{ title }}</h1>

    <button
      [routerLink]="['/basic/counter']"
      routerLinkActive="router-link-active"
      [routerLinkActiveOptions]="{exact: true}"
    >
      baasic counter
    </button>

    <button
      [routerLink]="['/basic/counter/30']"
      routerLinkActive="router-link-active"
      [routerLinkActiveOptions]="{exact: true}"
    >
      baasic counter param
    </button>

    <button
      [routerLink]="['/basic/pokemmon']"
      routerLinkActive="router-link-active"
    >
      baasic request pokemon
    </button>

    <button
      [routerLink]="['/basic/family']"
      routerLinkActive="router-link-active"
    >
      baasic input output father son
    </button>


    <router-outlet></router-outlet>
  `,
  styleUrl: './app.component.scss'
})
export class AppComponent {
  title = 'angular-tests';
}
