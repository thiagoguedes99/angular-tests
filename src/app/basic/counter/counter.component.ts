import { Component, OnInit } from '@angular/core';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'app-counter',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './counter.component.html',
  styleUrl: './counter.component.scss'
})
export class CounterComponent implements OnInit {
  counter: number = 10;

  constructor() { }

  ngOnInit(): void {
  }

  increaseBy( value: number ) {
    this.counter += value;
    // TODO:
    // console.log({ newValue: this.counter });

  }

}
