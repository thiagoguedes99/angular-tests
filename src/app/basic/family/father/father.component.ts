import { Component } from '@angular/core';
import { JsonPipe } from '@angular/common';

import { SonComponent } from '../son/son.component';
import { Client } from '../interfaces/client';

@Component({
  selector: 'app-father',
  standalone: true,
  imports: [SonComponent, JsonPipe],
  templateUrl: './father.component.html',
  styleUrl: './father.component.scss'
})
export class FatherComponent {
  public client?: Client;

  onSetClient( name: string ) {
    this.client = { id: 1, name };
  }

}
