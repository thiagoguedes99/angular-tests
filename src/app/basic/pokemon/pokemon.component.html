@if (charizard) {
  <h3>{{ charizard?.name | titlecase }}</h3>
  <!-- <h3>{{ charizard?.name }}</h3> -->
  <img [src]="charizard?.sprites?.front_default" [alt]="charizard?.name">
}
@else {
  <h2>Loading...</h2>
}
