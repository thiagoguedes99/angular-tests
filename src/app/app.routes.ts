import { Routes } from '@angular/router';
import { CounterComponent } from './basic/counter/counter.component';
import { PokemonComponent } from './basic/pokemon/pokemon.component';
import { FatherComponent } from './basic/family/father/father.component';

export const routes: Routes = [
  {
    path: 'basic/counter',
    component: CounterComponent
  },
  {
    path: 'basic/pokemmon',
    component: PokemonComponent
  },
  {
    path: 'basic/family',
    component: FatherComponent
  },
  // {
  //   path: '**',
  //   redirectTo: 'basic/family'
  // }

];
