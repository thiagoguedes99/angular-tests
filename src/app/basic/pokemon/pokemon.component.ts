import { Component, OnInit, inject } from '@angular/core';
import { TitleCasePipe } from '@angular/common';

import { PokemonServiceService } from './service/pokemon-service.service';
import { Pokemon } from './interfaces/pokemon';

@Component({
  selector: 'app-pokemon',
  standalone: true,
  // imports: [CommonModule],
  imports: [TitleCasePipe],
  providers: [PokemonServiceService],
  templateUrl: './pokemon.component.html',
  styleUrl: './pokemon.component.scss'
})
export class PokemonComponent implements OnInit {
  pokemonService: PokemonServiceService  = inject(PokemonServiceService);

  public charizard?: Pokemon;

  ngOnInit(): void {
    this.pokemonService.getPokemon(6)
      .subscribe( pokemon => {
        this.charizard = pokemon;
      });
  }

}
