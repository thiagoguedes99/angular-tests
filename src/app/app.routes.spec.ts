import { routes } from './app.routes'
import { CounterComponent } from './basic/counter/counter.component';
import { FatherComponent } from './basic/family/father/father.component';
import { PokemonComponent } from './basic/pokemon/pokemon.component';


describe('AppRoutes', () => {

  it('should to match snapshot', () => {
    expect(routes).toMatchSnapshot();
  });

  it('should to length routes', () => {
    expect(routes.length).toBe(3);
  });

  it('should to router for path basic/counter the CounterComponent', () => {
    expect(routes[0].path).toBe('basic/counter');
    expect(routes[0].component).toBe(CounterComponent);
  });

  it('should to router for path basic/pokemmon the PokemonComponent', () => {
    expect(routes[1].path).toBe('basic/pokemmon');
    expect(routes[1].component).toBe(PokemonComponent);
  });

  it('should to router for path basic/family the PokemonComponent', () => {
    expect(routes[2].path).toBe('basic/family');
    expect(routes[2].component).toBe(FatherComponent);
  });


});
