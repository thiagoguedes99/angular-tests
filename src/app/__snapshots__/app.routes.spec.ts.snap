// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoutes should to match snapshot 1`] = `
[
  {
    "component": [Function],
    "path": "basic/counter",
  },
  {
    "component": [Function],
    "path": "basic/pokemmon",
  },
  {
    "component": [Function],
    "path": "basic/family",
  },
]
`;
